packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +325,73 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	itExpects(
+		"full initialization of data object should not happen by default",
+		[
+			{
+				eventName: "fluid:telemetry:Container:Request_cancel",
+				error: "Non interactive/summarizer client's data object should not be initialized",
+			},
+		],
+		async () => {
+			const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+			const dataStoreFactory1 = new DataObjectFactory(
+				TestDataObjectType1,
+				TestDataObject1,
+				[],
+				[],
+			);
+			const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+				[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+			]);
+			const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+				dataStoreFactory1,
+				registryStoreEntries,
+				undefined,
+				[],
+			);
+
+			// Create a container for the first client.
+			const container1 = await provider.createContainer(runtimeFactory);
+			await assert.doesNotReject(
+				requestFluidObject<TestDataObject1>(container1, "/"),
+				"Initial creation of container and data store should succeed.",
+			);
+
+			// Create a summarizer for the container and do a summary shouldn't throw.
+			const createSummarizerResult = await createSummarizerFromFactory(
+				provider,
+				container1,
+				dataStoreFactory1,
+				undefined,
+				ContainerRuntimeFactoryWithDefaultDataStore,
+				registryStoreEntries,
+			);
+			const summarizer = createSummarizerResult.summarizer;
+			assert.doesNotThrow(
+				async () => summarizeNow(summarizer, "test"),
+				"Summarizing should not throw",
+			);
+
+			// In summarizer, load the data store should fail.
+			await assert.rejects(
+				async () =>
+					requestFluidObject<TestDataObject1>(createSummarizerResult.container, "/"),
+				(e) =>
+					e.message ===
+					"Non interactive/summarizer client's data object should not be initialized",
+				"This validates that a summarizer client doesn't fully initialize the data object.",
+			);
+
+			// Load second container, load the data store will also call initializingFromExisting and succeed.
+			const container2 = await provider.loadContainer(runtimeFactory);
+			await assert.doesNotReject(
+				async () => requestFluidObject<TestDataObject1>(container2, "/"),
+				"This validates that an interactive client doesn't fully initialize the data object.",
```

Comment:
Similar rationale as above, this seems more descriptive to someone that sees the test failing.
```suggestion
				"Loading data store in second interactive client should not throw.",
```
Also worth pointing out that in this case the data object *is* getting fully initialized, because our error is only thrown for non-interactive (aka summarizer) clients :).